File: ECE39595Homeworks/HW14Assigned/rkothand/DecoyDuck.java
public class DecoyDuck extends Duck {

   public DecoyDuck( ) {
      flyBehavior = new FlyNoWay( );
      quackBehavior = new MuteQuack( );
      laysEggsBehavior = new DoesNotLayEggs( );

   }

   public void display( ) {
      System.out.println("I’m a decoy duck");
   }
}

File: ECE39595Homeworks/HW9Assigned/Unique/DecoyDuck.cpp
#include <iostream>
#include "DecoyDuck.h"
#include "FlyNoWay.h"
#include "MuteQuack.h"
#include <memory>
using namespace std;

DecoyDuck::DecoyDuck( ) {
   flyBehavior = unique_ptr<FlyNoWay>(new FlyNoWay( ));
   quackBehavior = unique_ptr<MuteQuack>(new MuteQuack( ));
}

void DecoyDuck::display( ) {
   std::cout << "I’m a decoy duck" << std::endl;
}
